apps/client/src/components/Questions.js

```diff
@@ -91,20 +103,29 @@ const Questions = ({
         }
       });
     }
-  }, [checker, isActive, questionIndex, onQuestionNext, skipAnsweredQuestions]);
 
-  const onGoToQuestion = useCallback(
-    (questionId) => {
-      // Checker rewinding also needs to work when you already have a conlusion
-      // Go to the specific question in the stack
-      setActiveState("questions");
-      setFinishedState(["conclusion", "questions"], false);
-      setFinishedState("locationResult", true);
+    // @TODO: Refactor this code and move to checker.js
+    // Bug fix in case of refresh: hide already future answered questions (caused by setQuestionAnswers() in withChecker)
+    if (!contactConclusion) {
+      checker.stack.forEach((q, i) => {
+        if (checker.needContactExit(q)) {
+          // Set questionIndex to question that causes the needContactExit()
```

Comment:
Can you add the reason for setting the questionIndex in this case?

Reply:
Thanks